### PR TITLE
Deliveries

### PR DESCRIPTION
**Pull request by baldys**
### Description
Endpoints for deliveries:
create, index, show, update, delete

### CODE REVIEW

**Comment by tractionguest-integrations** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  You don't seem to have any wrike tickets for this PR!
    You need to add your wrike ticket to one of your commits, e.g., `[#1234] Your commit message`
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  Remember to use the `Squash and Merge` button to merge!
    Always remember, your final commit message needs to start with a wrike ticket like, `[#1234] Your commit message`.
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  <a href="https://stoplight.io/p/docs/gh/tractionguest/guest-api/openapi.yml?srn=gh/tractionguest/guest-api/openapi.yml&group=deliveries" target=_blank>View docs for this page</a>
  </td>
    </tr>
  </tbody>
</table>


<details>
<summary>Got false positives?</summary>

<p>Make changes to this repo's spellcheck function call in the dangerfile.</p>
</details>



### Typos for <a href="https://github.com/tractionguest/guest-api/blob/deliveries/guest-dart-sdk/docs/PackagesApi.md">PackagesApi.md</a>

| Line | Typo |
| ---- | ---- |
1 | # **guest_sdk.api.PackagesApi**
8 | All **URIs** are relative to *https://trac
12 | &#91;****createPackage****](PackagesApi.md#createPacka
13 | &#91;****getPackages****](PackagesApi.md#getPackages
16 | # ****createPackage****
17 | > Package **createPackage**(packageCreateParams)
17 | > Package createPackage(**packageCreateParams**)
42 |  ****packageCreateParams**** | &#91;**PackageCreateParams**]
42 |  **packageCreateParams** | &#91;****PackageCreateParams****](PackageCreateParams.md)| P
50 | &#91;**TractionGuestAuth**](../README.md#TractionGuestAu
54 | **Content-Type**: application/**json**
55 |  - **Accept**: application/**json**
59 | # ****getPackages****
60 | > **PaginatedPackagesList** getPackages(locationIds, limi
60 | > PaginatedPackagesList **getPackages**(locationIds, limit, offset)
60 | natedPackagesList getPackages(**locationIds**, limit, offset)
87 |  ****locationIds**** | **String**| A comma separ
88 |  **limit** | ****int****| Limits the results to a sp
89 |  **offset** | ****int****| Offsets the results to a s
93 | &#91;****PaginatedPackagesList****](PaginatedPackagesList.md)
97 | &#91;**TractionGuestAuth**](../README.md#TractionGuestAu
102 |  - **Accept**: application/**json**
      


### Typos for <a href="https://github.com/tractionguest/guest-api/blob/deliveries/guest-dart-sdk/docs/Package.md">Package.md</a>

| Line | Typo |
| ---- | ---- |
1 | # **guest_sdk.model.Package**
11 | **id** | ****int**** |  | &#91;default to null]
14 | ****packageState**** | **String** |  | &#91;optional
15 | ****carrierName**** | **String** |  | &#91;optional
16 | ****pickedUpAt**** | &#91;**DateTime**](DateTime.m
16 | **pickedUpAt** | &#91;****DateTime****](DateTime.md) |  | [optiona
17 | ****createdAt**** | &#91;**DateTime**](DateTime.m
17 | **createdAt** | &#91;****DateTime****](DateTime.md) |  | [default
19 | ****nlpResults**** | **String** |  | &#91;optional
20 | ****ocrResult**** | **String** |  | &#91;optional
      


### Typos for <a href="https://github.com/tractionguest/guest-api/blob/deliveries/guest-dart-sdk/docs/PaginatedPackagesList.md">PaginatedPackagesList.md</a>

| Line | Typo |
| ---- | ---- |
1 | # **guest_sdk.model.PaginatedPackagesList**
      


### Typos for <a href="https://github.com/tractionguest/guest-api/blob/deliveries/guest-dart-sdk/docs/PackageCreateParams.md">PackageCreateParams.md</a>

| Line | Typo |
| ---- | ---- |
1 | # **guest_sdk.model.PackageCreateParams**
11 | ****base64Image**** | **String** | Base64 encod
11 | *base64Image** | **String** | **Base64** encoded image on which to per
12 | ****locationId**** | **int** | Location id for
12 | **locationId** | ****int**** | Location id for where the
      


### Typos for <a href="https://github.com/tractionguest/guest-api/blob/deliveries/guest-dart-sdk/docs/Image.md">Image.md</a>

| Line | Typo |
| ---- | ---- |
1 | # **guest_sdk.model.Image**
11 | ****fullSizeUrl**** | **String** |  | &#91;optional
      


### Typos for <a href="https://github.com/tractionguest/guest-api/blob/deliveries/guest-dart-sdk/README.md">README.md</a>

| Line | Typo |
| ---- | ---- |
1 | # **guest_sdk**
2 | ative. Only paid accounts are **elligible**.  # Slack  Do you have Slack?
2 | nd will be removed in version **X.X.X**, which is scheduled for relea
2 | duled for release on or after **20XX-01-01**.\" Please note that the depre
4 | tomatically generated by the &#91;**OpenAPI** Generator](https://openapi-ge
7 | - Build package: **org.openapitools.codegen.languages.DartClientCodegen**
17 | ease include the following in **pubspec.yaml**
29 | ease include the following in **pubspec.yaml**
62 | All **URIs** are relative to *https://trac
66 | ***AuditLogsApi*** | &#91;**getAuditLog**](docs//Au
66 | *AuditLogsApi* | &#91;****getAuditLog****](docs//AuditLogsApi.md#geta
66 | pi.md#getauditlog) | **GET** /**audit_logs**/{audit_log_id} | Get an Audit
66 | itlog) | **GET** /audit_logs/{**audit_log_id**} | Get an AuditLog
66 | _logs/{audit_log_id} | Get an **AuditLog**
67 | ***AuditLogsApi*** | &#91;**getAuditLogs**](docs//A
67 | *AuditLogsApi* | &#91;****getAuditLogs****](docs//AuditLogsApi.md#geta
67 | i.md#getauditlogs) | **GET** /**audit_logs** | List all AuditLogs
67 | *GET** /audit_logs | List all **AuditLogs**
68 | ***CapacitiesApi*** | &#91;**getCapacity**](docs//Ca
68 | *CapacitiesApi* | &#91;****getCapacity****](docs//CapacitiesApi.md#get
68 | pacity) | **GET** /locations/{**location_id**}/capacities | Get the current
69 | ***CapacitiesApi*** | &#91;**getCapacityForecast**](
69 | *CapacitiesApi* | &#91;****getCapacityForecast****](docs//CapacitiesApi.md#get
69 | recast) | **GET** /locations/{**location_id**}/capacity_forecasts | Get the
69 | ET** /locations/{location_id}/**capacity_forecasts** | Get the capacity details fo
70 | ***EmailTemplatesApi*** | &#91;**getEmailTemplates**](do
70 | *EmailTemplatesApi* | &#91;****getEmailTemplates****](docs//EmailTemplatesApi.md
70 | getemailtemplates) | **GET** /**email_templates** | List all EmailTemplates
70 | * /email_templates | List all **EmailTemplates**
71 | ***GuestAlertsApi*** | &#91;**createGuestAlert**](doc
71 | *GuestAlertsApi* | &#91;****createGuestAlert****](docs//GuestAlertsApi.md#cr
71 | createguestalert) | **POST** /**guest_alerts** | Create GuestAlert
71 | POST** /guest_alerts | Create **GuestAlert**
72 | ***HostsApi*** | &#91;**createHost**](docs//Hos
72 | *HostsApi* | &#91;****createHost****](docs//HostsApi.md#createho
73 | ***HostsApi*** | &#91;**createHosts**](docs//Ho
73 | *HostsApi* | &#91;****createHosts****](docs//HostsApi.md#createho
74 | ***HostsApi*** | &#91;**getHosts**](docs//Hosts
74 | *HostsApi* | &#91;****getHosts****](docs//HostsApi.md#gethosts
75 | ***InvitesApi*** | &#91;**createLocationInvite**]
75 | *InvitesApi* | &#91;****createLocationInvite****](docs//InvitesApi.md#create
75 | nvite) | **POST** /locations/{**location_id**}/invites | Create an Invite
76 | ***InvitesApi*** | &#91;**deleteInvite**](docs//I
76 | *InvitesApi* | &#91;****deleteInvite****](docs//InvitesApi.md#delete
76 | nvite) | **DELETE** /invites/{**invite_id**} | Deletes an Invite
77 | ***InvitesApi*** | &#91;**getInvite**](docs//Invi
77 | *InvitesApi* | &#91;****getInvite****](docs//InvitesApi.md#getinv
77 | etinvite) | **GET** /invites/{**invite_id**} | Get a Invite
78 | ***InvitesApi*** | &#91;**getInvites**](docs//Inv
78 | *InvitesApi* | &#91;****getInvites****](docs//InvitesApi.md#getinv
79 | ***InvitesApi*** | &#91;**updateInvite**](docs//I
79 | *InvitesApi* | &#91;****updateInvite****](docs//InvitesApi.md#update
79 | teinvite) | **PUT** /invites/{**invite_id**} | Update a Invite
80 | ***LocationsApi*** | &#91;**getLocations**](docs//L
80 | *LocationsApi* | &#91;****getLocations****](docs//LocationsApi.md#getl
81 | ***LocationsApi*** | &#91;**getLocation**](docs//Lo
81 | *LocationsApi* | &#91;****getLocation****](docs//LocationsApi.md#getl
81 | cation) | **GET** /locations/{**location_id**} | Get the details of a locat
82 | ***PackagesApi*** | &#91;**createPackage**](docs//
82 | *PackagesApi* | &#91;****createPackage****](docs//PackagesApi.md#creat
83 | ***PackagesApi*** | &#91;**getPackages**](docs//Pa
83 | *PackagesApi* | &#91;****getPackages****](docs//PackagesApi.md#getpa
84 | ***RegistrationApi*** | &#91;**getRegistration**](docs
84 | *RegistrationApi* | &#91;****getRegistration****](docs//RegistrationApi.md#g
84 | on) | **GET** /registrations/{**registration_id**} | Get a Registration
85 | ***RegistrationApi*** | &#91;**getRegistrations**](doc
85 | *RegistrationApi* | &#91;****getRegistrations****](docs//RegistrationApi.md#g
86 | ***SigninsApi*** | &#91;**createSignin**](docs//S
86 | *SigninsApi* | &#91;****createSignin****](docs//SigninsApi.md#create
86 | .md#createsignin) | **POST** /**signins** | Create Signin
86 |  | **POST** /signins | Create **Signin**
87 | ***SigninsApi*** | &#91;**getSignin**](docs//Sign
87 | *SigninsApi* | &#91;****getSignin****](docs//SigninsApi.md#getsig
87 | sApi.md#getsignin) | **GET** /**signins**/{signin_id} | Get a Signin
87 | etsignin) | **GET** /signins/{**signin_id**} | Get a Signin
87 |  /signins/{signin_id} | Get a **Signin**
88 | ***SigninsApi*** | &#91;**getSignins**](docs//Sig
88 | *SigninsApi* | &#91;****getSignins****](docs//SigninsApi.md#getsig
88 | Api.md#getsignins) | **GET** /**signins** | List all Signins
88 | | **GET** /signins | List all **Signins**
89 | ***SigninsApi*** | &#91;**updateSignin**](docs//S
89 | *SigninsApi* | &#91;****updateSignin****](docs//SigninsApi.md#update
89 | i.md#updatesignin) | **PUT** /**signins**/{signin_id} | Update a Signin
89 | tesignin) | **PUT** /signins/{**signin_id**} | Update a Signin
89 | ignins/{signin_id} | Update a **Signin**
90 | ***UsersApi*** | &#91;**getCurrentUser**](docs/
90 | *UsersApi* | &#91;****getCurrentUser****](docs//UsersApi.md#getcurre
90 | urrentuser) | **GET** /users/{**user_id**} | Get the current User
91 | ***WatchlistsApi*** | &#91;**createWatchlist**](docs
91 | *WatchlistsApi* | &#91;****createWatchlist****](docs//WatchlistsApi.md#cre
92 | ***WatchlistsApi*** | &#91;**deleteWatchlist**](docs
92 | *WatchlistsApi* | &#91;****deleteWatchlist****](docs//WatchlistsApi.md#del
92 | st) | **DELETE** /watchlists/{**watchlist_id**} | Deletes a Watchlist
93 | ***WatchlistsApi*** | &#91;**getWatchlist**](docs//W
93 | *WatchlistsApi* | &#91;****getWatchlist****](docs//WatchlistsApi.md#get
93 | hlist) | **GET** /watchlists/{**watchlist_id**} | Get a Watchlist
94 | ***WatchlistsApi*** | &#91;**getWatchlists**](docs//
94 | *WatchlistsApi* | &#91;****getWatchlists****](docs//WatchlistsApi.md#get
95 | ***WatchlistsApi*** | &#91;**updateWatchlist**](docs
95 | *WatchlistsApi* | &#91;****updateWatchlist****](docs//WatchlistsApi.md#upd
95 | hlist) | **PUT** /watchlists/{**watchlist_id**} | Update a Watchlist
100 |  - &#91;**AuditLog**](docs//AuditLog.md)
101 |  - &#91;**AuditLogChange**](docs//AuditLogChange.md)
102 |  - &#91;**BatchJob**](docs//BatchJob.md)
104 |  - &#91;**CapacityByHourResponse**](docs//CapacityByHourResponse
105 |  - &#91;**CapacityForecast**](docs//CapacityForecast.md)
106 |  - &#91;**CustomField**](docs//CustomField.md)
107 |  - &#91;**Docusign**](docs//Docusign.md)
108 |  - &#91;**EmailTemplate**](docs//EmailTemplate.md)
110 |  - &#91;**ErrorsList**](docs//ErrorsList.md)
111 |  - &#91;**ExternalWatchlistResult**](docs//ExternalWatchlistResul
112 |  - &#91;**GroupVisit**](docs//GroupVisit.md)
113 |  - &#91;**GuestAlertCreateParams**](docs//GuestAlertCreateParams
114 |  - &#91;**GuestAlertDetail**](docs//GuestAlertDetail.md)
115 |  - &#91;**GuestAlertSigninSelectors**](docs//GuestAlertSigninSelect
116 |  - &#91;**GuestResponse**](docs//GuestResponse.md)
118 |  - &#91;**HostBatchCreateParams**](docs//HostBatchCreateParams.
119 |  - &#91;**HostCreateParams**](docs//HostCreateParams.md)
121 |  - &#91;**InternalWatchlistResult**](docs//InternalWatchlistResul
123 |  - &#91;**InviteCreateParams**](docs//InviteCreateParams.md)
124 |  - &#91;**InviteDetail**](docs//InviteDetail.md)
125 |  - &#91;**InviteUpdateParams**](docs//InviteUpdateParams.md)
126 |  - &#91;**InviteWatchlist**](docs//InviteWatchlist.md)
128 |  - &#91;**NotificationTrigger**](docs//NotificationTrigger.md
129 |  - &#91;**NotificationTriggerCreateParams**](docs//NotificationTriggerCre
131 |  - &#91;**PackageCreateParams**](docs//PackageCreateParams.md
132 |  - &#91;**PaginatedAuditLogsList**](docs//PaginatedAuditLogsList
133 |  - &#91;**PaginatedEmailTemplatesList**](docs//PaginatedEmailTemplate
134 |  - &#91;**PaginatedHostsList**](docs//PaginatedHostsList.md)
135 |  - &#91;**PaginatedInvitesList**](docs//PaginatedInvitesList.m
136 |  - &#91;**PaginatedLocationsList**](docs//PaginatedLocationsList
137 |  - &#91;**PaginatedPackagesList**](docs//PaginatedPackagesList.
138 |  - &#91;**PaginatedRegistrationsList**](docs//PaginatedRegistrations
139 |  - &#91;**PaginatedSigninsList**](docs//PaginatedSigninsList.m
140 |  - &#91;**PaginatedWatchlistList**](docs//PaginatedWatchlistList
142 |  - &#91;**PermissionGroup**](docs//PermissionGroup.md)
144 |  - &#91;**RegistrationDetail**](docs//RegistrationDetail.md)
145 |  - &#91;**SignableDocument**](docs//SignableDocument.md)
146 |  - &#91;**Signin**](docs//Signin.md)
147 |  - &#91;**SigninAcknowledgement**](docs//SigninAcknowledgement.
148 |  - &#91;**SigninCreateParams**](docs//SigninCreateParams.md)
149 |  - &#91;**SigninData**](docs//SigninData.md)
150 |  - &#91;**SigninDetail**](docs//SigninDetail.md)
151 |  - &#91;**SigninUpdateParams**](docs//SigninUpdateParams.md)
152 |  - &#91;**SigninWatchlist**](docs//SigninWatchlist.md)
153 |  - &#91;**SimpleSignature**](docs//SimpleSignature.md)
157 |  - &#91;**WatchlistCreateParams**](docs//WatchlistCreateParams.
158 |  - &#91;**WatchlistMatch**](docs//WatchlistMatch.md)
159 |  - &#91;**WatchlistSearch**](docs//WatchlistSearch.md)
165 | ## **TractionGuestAuth**
      
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 8417d1d40a44714f23b822382efc1374dbb5758b
</p>

**Comment by brandonjmckay** [comment]
You're going to want to change the base of this PR to `develop` now, we changed some of the process since this branch was first made

File: models/Host.v1.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 title: Host
-description: A client host assigned to a Signin
+description: "The root of the Host type's schema."
```
Comment (brandonjmckay) [inline models/Host.v1.yaml:2]: Should revert this file's changes
```suggestion
description: A client host assigned to a Signin
```

File: models/Package.v1.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    description: 'A carrier name that gets detected on the shipping label. i.e. USPS, Purolator, DHL, Canada Post, Royal Mail, etc... '
+  picked_up_at:
+    type: string
+    format: date-time
+    nullable: true
+  created_at:
+    type: string
+    format: date-time
+  image:
+    $ref: ./Image.v1.yaml
```
Comment (brandonjmckay) [inline models/Package.v1.yaml:73]: why even have this as a separate model?
Seems like it's only the one field on it there

File: models/PackageCreateParams.v1.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+title: PackageCreateParams.v1
+type: object
+description: "Root of a PackageCreateParams type's schema. \n[base64_image] string is provided, intented to be an encoded image of a shipping label. The image will be processed to determine the package's intended recipient. If a match is found between the recipient and an existing host, they will subsequently be notified about the arrival of their package, at the [location_id] specified. \n\n"
```
Comment (brandonjmckay) [inline models/PackageCreateParams.v1.yaml:3]: 🌵 Not sure why there was extra line endings at the end, and some parts seemed wordy
```suggestion
description: "A [base64_image] string is provided as an encoded image of a shipping label. The image will be processed to determine the package's intended recipient. If a match is found between the recipient and an existing host, they'll be notified about the arrival of their package at the [location_id] specified."
```

File: openapi.yml [hunk 4]
```diff
@@ -3683,6 +3683,273 @@ paths:
...
+                $ref: ./models/PaginatedPackagesList.v1.yaml
+              examples:
+                example:
+                  value:
+                    packages:
+                      - id: 70
+                        package_state: picked_up
+                        recipient:
+                          id: 5
+                          email: tsala@tractionguest.com
```
Comment (brandonjmckay) [inline openapi.yml:3707]: not sure we should use real emails in the public API unless we're wanting people to contact them?

File: openapi.yml [hunk 5]
```diff
@@ -3683,6 +3683,273 @@ paths:
...
+                    package_state: PROCESSING
+                    recipient: null
+                    location:
+                      name: Default
+                      id: 1
+                    carrier_name: null
+                    created_at: '2018-02-10T09:30Z'
+                    picked_up_at: null
+                    image_url: some text
+        '400':
```
Comment (brandonjmckay) [inline openapi.yml:3851]: should use 4xx too

File: guest-dart-sdk/docs/Image.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
+# guest_sdk.model.Image
```
Comment (brandonjmckay) [inline guest-dart-sdk/docs/Image.md:1]: There's no description to describe what's the plan with this PR, so I don't know why you'd manually change the dart SDK at all.

What's happening here?
Are you manually creating a new SDK? Seems like that should be a whole different PR

File: models/Host.v1.yaml (hunk 1 above)
Comment (baldys) [reply models/Host.v1.yaml:2]: Is `A client host assigned to a Signin` an appropriate description for a Host?

File: models/PackageUpdateParams.v1.yaml [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  [carrier_name] - change/update the package's carrier/courier information
+properties:
+  picked_up:
+    type: boolean
+    description: changes the package_state to picked up and assigns non null value to picked_up_at
+  carrier_name:
+    type: string
+    description: "change/update the package's carrier/courier information"
+  recipient_id:
+    type: integer
```
Comment (brandonjmckay) [inline models/PackageUpdateParams.v1.yaml:26]: Didn't we talk about needing/wanting to make all IDs strings?
That way we could transition to using UUIDs without an issue at some point.
@stevenallen05

File: openapi.yml [hunk 8]
```diff
@@ -3660,6 +3660,273 @@ paths:
...
+                        image:
+                          full_size_url: 'https://i.pravatar.cc/'
+                      - id: 74
+                        package_state: recipient_matched
+                        recipient:
+                          id: 4
+                          email: vbaldys@example.com
+                          first_name: Veronica
+                          last_name: Baldys
+                          mobile_number: (604)999-5982
```
Comment (brandonjmckay) [inline openapi.yml:3738]: fake, right?

Comment (fernandojinzenji) [reply openapi.yml:3738]: Yeah, I really suggest you change this in case it is your phone number since it will be part of our API documentation. Unless you want to be part of the CS team to answer customer questions.

File: models/PackageCreateParams.v1.yaml [hunk 9]
```diff
@@ -0,0 +1,20 @@
+title: PackageCreateParams.v1
+type: object
+description: "A [base64_image] string is provided as an encoded image of a shipping label. The image will be processed to determine the package's intended recipient. If a match is found between the recipient and an existing host, they'll be notified about the arrival of their package at the [location_id] specified."
```
Comment (stevenallen05) [inline models/PackageCreateParams.v1.yaml:3]: Nice description! 👍

File: models/PackageUpdateParams.v1.yaml (hunk 7 above)
Comment (stevenallen05) [reply models/PackageUpdateParams.v1.yaml:26]: Yes, this should be a string. But it's also too late in the application development to turn that into a rule right now. Let's discuss next week about making that part of the styleguide.
